### PR TITLE
fix COLR font check being racy

### PR DESCRIPTION
**Pull request by bwindels**
also make sure it doesn't run more than once.
keeping the FF sniffing because missing "extract canvas data" permissions
would still break the check.

### CODE REVIEW

**Comment by ara4n** [comment]
the point of the sleep is to let the font load inside the svg, not the image itself load, which apparently is a safari issue according to the original

**Comment by ara4n** [comment]
that said, you could always do a UA check for Safari to mitigate that

**Comment by bwindels** [comment]
> the point of the sleep is to let the font load inside the svg, not the image itself load, which apparently is a safari issue according to the original

Hmm, good point. I'd be inclined to test it though, as, as it stands, it is going to break on any slow connection. FF for me always fails the check for example (as font fetching is de-prioritized perhaps?) so wonder how well tested the chromecheck code is ...

File: src/utils/FontManager.js [hunk 1]
```diff
@@ -21,21 +21,43 @@ limitations under the License.
...
+        if (safariVersionMatch) {
+            const macOSVersionStr = safariVersionMatch[1];
+            const safariVersionStr = safariVersionMatch[2];
+            const macOSVersion = macOSVersionStr.split("_").map(n => parseInt(n, 10));
+            const safariVersion = safariVersionStr.split(".").map(n => parseInt(n, 10));
+            const colrFontSupported = macOSVersion[0] >= 10 && macOSVersion[1] >= 14 && safariVersion[0] >= 12;
+            // https://www.colorfonts.wtf/ states safari supports COLR fonts from this version on
+            console.log(`Browser is Safari - requiring macOS 10.14 and Safari 12,` +
+                `detected Safari ${safariVersionStr} on macOS ${macOSVersionStr},` +
+                `supported: ${colrFontSupported}`);
```
Comment (jryans) [inline src/utils/FontManager.js]: Maybe add the word "COLR" in this log message for easy filtering?

```suggestion
                `COLR supported: ${colrFontSupported}`);
```

File: src/utils/FontManager.js [hunk 2]
```diff
@@ -21,21 +21,43 @@ limitations under the License.
...
+    try {
+        const safariVersionMatch = ua.match(/Mac OS X ([\d|_]+).*Version\/([\d|\.]+) Safari/);
+        if (safariVersionMatch) {
+            const macOSVersionStr = safariVersionMatch[1];
+            const safariVersionStr = safariVersionMatch[2];
+            const macOSVersion = macOSVersionStr.split("_").map(n => parseInt(n, 10));
+            const safariVersion = safariVersionStr.split(".").map(n => parseInt(n, 10));
+            const colrFontSupported = macOSVersion[0] >= 10 && macOSVersion[1] >= 14 && safariVersion[0] >= 12;
+            // https://www.colorfonts.wtf/ states safari supports COLR fonts from this version on
+            console.log(`Browser is Safari - requiring macOS 10.14 and Safari 12,` +
```
Comment (jryans) [inline src/utils/FontManager.js]:
```suggestion
            console.log(`Browser is Safari - requiring macOS 10.14 and Safari 12, ` +
```

File: src/utils/FontManager.js [hunk 3]
```diff
@@ -21,21 +21,43 @@ limitations under the License.
...
+                `detected Safari ${safariVersionStr} on macOS ${macOSVersionStr},` +
```
Comment (jryans) [inline src/utils/FontManager.js]:
```suggestion
                `detected Safari ${safariVersionStr} on macOS ${macOSVersionStr}, ` +
```

**Pull request merged**